ipv-api/src/main/java/uk/gov/di/authentication/ipv/helpers/IPVCallbackHelper.java

```diff
@@ -183,24 +184,28 @@ public void doAccountIntervention(AccountInterventionStatus accountInterventionS
         }
     }
 
-    public Optional<ErrorObject> validateUserIdentityResponse(UserInfo userIdentityUserInfo)
+    public Optional<ErrorObject> validateUserIdentityResponse(
+            UserInfo userIdentityUserInfo, List<VectorOfTrust> vtrList)
             throws IpvCallbackException {
         LOG.info("Validating userinfo response");
-        if (!LevelOfConfidence.MEDIUM_LEVEL
-                .getValue()
-                .equals(userIdentityUserInfo.getClaim(VOT.getValue()))) {
-            LOG.warn("IPV missing vot or vot not P2.");
-            return Optional.of(OAuth2Error.ACCESS_DENIED);
+        for (VectorOfTrust vtr : vtrList) {
+            if (vtr.getLevelOfConfidence()
+                    .getValue()
+                    .equals(userIdentityUserInfo.getClaim(VOT.getValue()))) {
+                var trustmarkURL =
+                        buildURI(
+                                        configurationService.getOidcApiBaseURL().orElseThrow(),
+                                        "/trustmark")
+                                .toString();
+                if (!trustmarkURL.equals(userIdentityUserInfo.getClaim(VTM.getValue()))) {
+                    LOG.warn("VTM does not contain expected trustmark URL");
+                    throw new IpvCallbackException("IPV trustmark is invalid");
+                }
```

Comment:
Does this need to be inside the loop?

Reply:
It only gets called once (when a match is found) :)